client/src/pages/SignUp.js

```diff
@@ -0,0 +1,247 @@
+import React, { useState } from "react";
+import Button from "@material-ui/core/Button";
+import TextField from "@material-ui/core/TextField";
+import Typography from "@material-ui/core/Typography";
+import Modal from "react-modal";
+import { Link } from "react-router-dom";
+import { formStyles, modalStyles, signUpStyles } from "../themes/theme";
+
+Modal.setAppElement("#root");
+
+function SignUp() {
+  const [modalIsOpen, setModalIsOpen] = useState(true);
+  const [user, setUser] = useState({
+    name: "",
+    email: "",
+    password: "",
+  });
+  const [passwordTwo, setPasswordTwo] = useState("");
+  const [error, setError] = useState({
+    name: false,
+    email: false,
+    password: false,
+    passwordTwo: false,
+  });
+  const [errorMessage, setErrorMessage] = useState({
+    name: "",
+    email: "",
+    password: "",
+    passwordTwo: "",
+  });
+
+  // Handle text input
+  function handleChange(event) {
+    const field = event.target.name;
+    const tempUser = user;
+
+    // Reset error messages
+    const err = error;
+    const errorText = errorMessage;
+    err[field] = false;
+    errorText[field] = "";
+
+    // Check if we're entering in the password 2 field
+    if (field === "passwordTwo") {
```

Comment:
I'd recommend using a form package like Formik to handle form validation, a lot cleaner than having to do it yourself